compiler/noirc_frontend/src/parser/parser/expression.rs

```diff
@@ -553,9 +553,9 @@ impl<'a> Parser<'a> {
             if generics.is_empty() {
                 self.expected_token(Token::Less);
             }
-            generics
+            Some(generics)
```

Comment:
Can you change the `if self.eat_double_colon() {` above to `self.eat_double_colon().then(|| {` ? :cowboy_hat_face: